docs/hooks.md

```diff
@@ -0,0 +1,67 @@
+# Hooks
+
```

Comment:
It should be mentioned that this feature depends on the `xo-server-hooks` plugin.
Also we need to decide for which plan(s) it will be available.

And I think we should mark this API as experimental.